app/model/schema/messaging.py

```diff
@@ -16,9 +16,35 @@
 
 SPDX-License-Identifier: Apache-2.0
 """
+import re
+import string
 from typing import Optional
 
-from pydantic import BaseModel, EmailStr, Field, Json, conlist, constr
+from pydantic import (
+    BaseModel,
+    EmailStr,
+    Field,
+    Json,
+    conbytes,
+    conlist,
+    constr,
+    root_validator,
+    validator,
+)
+
+
+def _get_unprintable_ascii_chars() -> list[str]:
+    return [chr(c) for c in range(128) if chr(c) not in string.printable]
+
+
+UNPRINTABLE_ASCII_CHARS = tuple(_get_unprintable_ascii_chars())
+INVALID_PATH_CHARS = "".join(UNPRINTABLE_ASCII_CHARS)
+INVALID_FILENAME_CHARS = INVALID_PATH_CHARS + "/"
+INVALID_WIN_PATH_CHARS = INVALID_PATH_CHARS + ':*?"<>|\t\n\r\x0b\x0c'
+INVALID_WIN_FILENAME_CHARS = INVALID_FILENAME_CHARS + INVALID_WIN_PATH_CHARS + "\\"
+RE_INVALID_WIN_FILENAME = re.compile(
+    f"[{re.escape(INVALID_WIN_FILENAME_CHARS):s}]", re.UNICODE
+)
```

Comment:
It's hard to understand what each variable is, so I'd be happy if you could write a comment.